### PR TITLE
Improve esbulk with configurable exponential backoff

### PR DESCRIPTION
**Pull request by tgrk**
Besides testing exponential backoff I also include it for all ES API operations, exposed maximum number of retries as CLI argument and do basic refactoring of repetitive HTTP related code. More tests to come in follow up PRs.

Question is does it make sense to also expose backoff strategy as an CLI arg? WDYT?

### CODE REVIEW

File: indexing.go [hunk 1]
```diff
@@ -368,24 +349,92 @@ func CreateIndex(options Options) error {
...
+func PickServerURI(servers []string) string {
+	rand.Seed(time.Now().Unix())
+	return servers[rand.Intn(len(servers))]
+}
+
+// LogBackoffErrors logs in case backoff logic was triggered
+func LogBackoffErrors(clientLogString string) {
+	if clientLogString != "" {
+		// to make logging format consistent we need to temporary
+		// remove timestamp profix as it was already collected by
```
Comment (nkly) [inline indexing.go]: Typo `profix` -> `prefix`

File: indexing.go [hunk 2]
```diff
@@ -368,24 +349,92 @@ func CreateIndex(options Options) error {
...
+}
+
+// PickServerURI returns random server URL from available servers
+func PickServerURI(servers []string) string {
+	rand.Seed(time.Now().Unix())
+	return servers[rand.Intn(len(servers))]
+}
+
+// LogBackoffErrors logs in case backoff logic was triggered
+func LogBackoffErrors(clientLogString string) {
```
Comment (nkly) [inline indexing.go]: This fn doesn't seem to be used outside the package. I think it make sense to make it private.

**Review by nkly — APPROVED** [review]
Looks good :+1:

File: README.md [hunk 3]
```diff
@@ -66,6 +66,7 @@ Usage
       -w int
               number of workers to use (default 4)
       -z    unzip gz'd file on the fly
+      -retries    maximum number of backoff retries (default 3)
```
Comment (wmalik) [inline README.md:69]: maybe:
```
-retries    maximum number of retries (default 3) for HTTP requests
```

File: indexing.go [hunk 4]
```diff
@@ -368,24 +349,91 @@ func CreateIndex(options Options) error {
...
+	return req, nil
+}
+
+// PickServerURI returns random server URL from available servers
+func PickServerURI(servers []string) string {
+	rand.Seed(time.Now().Unix())
+	return servers[rand.Intn(len(servers))]
+}
+
+func logBackoffErrors(clientLogString string) {
```
Comment (wmalik) [inline indexing.go]: You could rename this function to `logClientErrors`?

**Review by wmalik — APPROVED** [review]
Looks good (minor comments)

**Pull request merged**